src/include/network/notifiable_task.h

```diff
@@ -59,20 +54,14 @@ class NotifiableTask {
    * Constructs a new NotifiableTask instance.
    * @param task_id a unique id assigned to this task
    */
-  explicit NotifiableTask(int task_id)
-      : task_id_(task_id) {
-    base_ = event_base_new();
-    // TODO(tianyu) Error handling here can be better perhaps?
-    if (base_ == nullptr) {
-      LOG_ERROR("Can't allocate event base\n");
-      exit(1);
-    }
+  explicit NotifiableTask(int task_id) : task_id_(task_id) {
+    base_ = EventUtil::EventBaseNew();
```

Comment:
Are we handling exceptions when we create `NotifiableTask` like `ConnectionDispatcherTask` or `ConnectionHandlerTask` now?

Reply:
They should both only interact with libevent through the NotifiableTask interface. (the base is private and not accessible to child classes)